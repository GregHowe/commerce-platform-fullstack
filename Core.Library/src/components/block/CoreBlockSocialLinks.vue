<template>
	<div class="block__body">
		<div class="block__header_group">
			<div
				v-if="blockHeadline"
				class="block__headline"
			>
				{{ blockHeadline }}
			</div>
		</div>
		<div class="block__icon_group">
			<div
				v-if="blockFacebook"
				class="block__facebook"
			>
				<a
					class="block__link"
					:href="blockFacebook"
				>
					<CoreIconFacebook />
				</a>
			</div>
			<div
				v-if="blockLinkedin"
				class="block__linkedin"
			>
				<a
					:href="blockLinkedin"
					class="block__link"
				>
					<CoreIconLinkedin />
				</a>
			</div>
			<div
				v-if="blockInstagram"
				class="block__instagram"
			>
				<a
					:href="blockInstagram"
					class="block__link"
				>
					<CoreIconInstagram />
				</a>
			</div>
			<div
				v-if="blockTwitter"
				class="block__twitter"
			>
				<a
					:href="blockTwitter"
					class="block__link"
				>
					<CoreIconTwitter />
				</a>
			</div>
			<div
				v-if="blockYoutube"
				class="block__youtube"
			>
				<a
					:href="blockYoutube"
					class="block__link"
				>
					<CoreIconYoutube />
				</a>
			</div>
		</div>
		<div class="block__header_group">
			<div
				v-if="blockUsername"
				class="block__username"
			>
				{{ blockUsername }}
			</div>
		</div>
	</div>
</template>

<script>
import { renderData } from "@libraryHelpers/dataComponents.js";

export default {
	name: "CoreBlockSocialLinks",
	props: {
		settings: {
			type: Object,
			required: true,
		},
		dataSite: {
			type: Object,
			default: () => {},
		},
	},
	computed: {
		blockHeadline() {
			return this.settings.headline || null;
		},
		blockUsername() {
			return this.settings.username || null;
		},
		blockFacebook() {
			return renderData(this.settings.facebook, this.dataSite);
		},
		blockLinkedin() {
			return renderData(this.settings.linkedin, this.dataSite);
		},
		blockInstagram() {
			return this.settings.instagram || "";
		},
		blockTwitter() {
			return renderData(this.settings.twitter, this.dataSite);
		},
		blockYoutube() {
			return this.settings.youtube || "";
		},
	},
};
</script>

<style lang="scss">
.block_social-links {
	.block__body {
		display: flex;
		flex-direction: column;
		justify-content: var(--_core__body_justify-content);
		padding: 1rem;
		background: transparent;
		color: var(--_core__container_color);

		.block__link {
			display: flex;
			flex-direction: column;
			justify-content: var(--_core__body_justify-content);
			margin: 0 0.5rem;
			height: 100%;
			color: var(--_core__container_color);
		}

		.block__icon_group {
			display: flex;
			flex-direction: row;
			justify-content: var(--_core__body_justify-content);
			text-align: var(--_core__body_text-align);
			padding: 0.8rem 0;
		}

		.block__header_group {
			display: flex;
			flex-direction: column;
			justify-content: var(--_core__body_justify-content);
			text-align: var(--_core__body_text-align);
		}
	}

	&.block_social-links_contrast .block__body {
		background: var(--_core__container_color);
		color: var(--core__color_light-alt);
		a {
			color: var(--core__color_light-alt);
		}
	}

	&.block_social-links_background .block__body {
		background: var(--_core__container_background);
		color: var(--_core__container_color);
	}
	&.block_social-links_slim {
		.block__body {
			background: transparent;
			padding: 0;
			.block__link {
				margin: 0 0.4rem;
				padding: 0;
				height: auto;
			}
			.block__icon_group {
				padding: 0;
			}
		}
		&.block_social-links_align-left .block__body {
			.block__link:first-child {
				margin-left: 0;
			}
		}
		&.block_social-links_align-right .block__body {
			.block__link:last-child {
				margin-right: 0;
			}
		}
	}
	&.block_social-links_align-left .block__body {
		align-items: flex-start;
	}
	&.block_social-links_align-right .block__body {
		align-items: flex-end;
	}
}
</style>
