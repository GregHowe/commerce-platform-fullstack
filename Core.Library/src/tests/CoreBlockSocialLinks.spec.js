import CoreBlockSocialLinks from "../components/block/CoreBlockSocialLinks";
import "@testing-library/jest-dom";
import { render } from "@testing-library/vue";

const renderComponent = async (propsData = {}) => {
	return render(CoreBlockSocialLinks, {
		propsData: {
			dataSite: {
				id: 12,
			},
			...propsData,
		},
	});
};

describe("CoreBlockSocialLinks", () => {
	test("if facebook value is set, make sure facebook icon appears and link matches value", async () => {
		const { container } = await renderComponent({
			settings: {
				facebook: "http://facebook.com",
			},
		});
		expect(container).toMatchSnapshot();
	});

	test("if youtube value is set, make sure youtube icon appears and link matches value", async () => {
		const { container } = await renderComponent({
			settings: {
				youtube: "http://youtube.com",
			},
		});
		expect(container).toMatchSnapshot();
	});

	test("if headline is set, make sure headline appears", async () => {
		const { container } = await renderComponent({
			settings: { headline: "text" },
		});
		expect(container).toMatchSnapshot();
	});

	test("if username is set, make sure username appears", async () => {
		const { container } = await renderComponent({
			settings: { username: "text" },
		});
		expect(container).toMatchSnapshot();
	});
});
