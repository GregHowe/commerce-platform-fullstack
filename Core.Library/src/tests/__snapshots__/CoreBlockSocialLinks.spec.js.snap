// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreBlockSocialLinks if facebook value is set, make sure facebook icon appears and link matches value 1`] = `
<div>
  <div
    class="block__body"
  >
    <div
      class="block__header_group"
    >
      <!---->
    </div>
     
    <div
      class="block__icon_group"
    >
      <div
        class="block__facebook"
      >
        <a
          class="block__link"
          href="http://facebook.com"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0C5.373 0 0 5.373 0 12C0 18.627 5.373 24 12 24C18.627 24 24 18.627 24 12C24 5.373 18.627 0 12 0ZM15 8H13.65C13.112 8 13 8.221 13 8.778V10H15L14.791 12H13V19H10V12H8V10H10V7.692C10 5.923 10.931 5 13.029 5H15V8Z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
       
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
    </div>
     
    <div
      class="block__header_group"
    >
      <!---->
    </div>
  </div>
</div>
`;

exports[`CoreBlockSocialLinks if headline is set, make sure headline appears 1`] = `
<div>
  <div
    class="block__body"
  >
    <div
      class="block__header_group"
    >
      <div
        class="block__headline"
      >
        
			text
		
      </div>
    </div>
     
    <div
      class="block__icon_group"
    >
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
    </div>
     
    <div
      class="block__header_group"
    >
      <!---->
    </div>
  </div>
</div>
`;

exports[`CoreBlockSocialLinks if username is set, make sure username appears 1`] = `
<div>
  <div
    class="block__body"
  >
    <div
      class="block__header_group"
    >
      <!---->
    </div>
     
    <div
      class="block__icon_group"
    >
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
    </div>
     
    <div
      class="block__header_group"
    >
      <div
        class="block__username"
      >
        
			text
		
      </div>
    </div>
  </div>
</div>
`;

exports[`CoreBlockSocialLinks if youtube value is set, make sure youtube icon appears and link matches value 1`] = `
<div>
  <div
    class="block__body"
  >
    <div
      class="block__header_group"
    >
      <!---->
    </div>
     
    <div
      class="block__icon_group"
    >
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
       
      <div
        class="block__youtube"
      >
        <a
          class="block__link"
          href="http://youtube.com"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0C5.373 0 0 5.373 0 12C0 18.627 5.373 24 12 24C18.627 24 24 18.627 24 12C24 5.373 18.627 0 12 0ZM16.441 16.892C14.339 17.036 9.657 17.036 7.558 16.892C5.282 16.736 5.017 15.622 5 12C5.017 8.371 5.285 7.264 7.558 7.108C9.657 6.964 14.34 6.964 16.441 7.108C18.718 7.264 18.982 8.378 19 12C18.982 15.629 18.715 16.736 16.441 16.892ZM10 9.658L14.917 11.996L10 14.342V9.658Z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </div>
     
    <div
      class="block__header_group"
    >
      <!---->
    </div>
  </div>
</div>
`;
